src/js/relayclient/ServerHelper.js

```diff
@@ -107,24 +111,66 @@ class ActiveRelayPinger {
 
 class ServerHelper {
 
-    constructor(httpSend, { verbose, minStake, minDelay, relayFilter, relayComparator }) {
+    constructor( httpSend, failedRelays,
+            {
+                verbose,
+                minStake, minDelay, //params for relayFilter: filter out this relay if unstakeDelay or stake are too low.
+                relayTimeoutGrace,  //ignore score drop of a relay after this time (seconds)
+                calculateRelayScore, //function: return relay score, higher the better. default uses transactionFee and some randomness
+                relayFilter,        //function: return false to filter out a relay. default uses minStake, minDelay
+                addScoreRandomness,  //function: return Math.random (0..1), to fairly distribute among relays with same score.
+                                     // (used by test to REMOVE the randomness, and make the test deterministic.
+            }) {
+
         this.httpSend = httpSend
         this.verbose = verbose
-        
+        this.failedRelays = failedRelays
+        this.relayTimeoutGrace = relayTimeoutGrace || DEFAULT_RELAY_TIMEOUT_GRACE_SEC
+
+        this.addScoreRandomness = addScoreRandomness || Math.random
+
+        this.calculateRelayScore = calculateRelayScore || this.defaultCalculateRelayScore.bind(this)
+
+        //default filter: either calculateRelayScore didn't set "score" field,
+        // or if unstakeDelay is below min, or if stake is below min.
         this.relayFilter = relayFilter || ((relay) => (
```

Comment:
Could you please make it a function, just as "calculateRelayScore", for consistency and stuff?

Reply:
requires too much work... need to save the config params as members to be used by another fucntion..